### ISSUE TITLE
Codec needed to cache (Seq of) custom types in Caffeine cache

### ISSUE THREAD

**Issue by sslavic** [cb372/scalacache#90]
After upgrading from 0.7.5 to 0.8.0 of scalacache, compilation of

```
import scalacache._
import caffeine._

implicit val scalaCache = ScalaCache(CaffeineCache())

def baz(): Seq[foo.Bar] = ...

def logic(): [foo.Bar] = sync.cachingWithTTL(key)(ttl)(baz)
```

fails at cachingWithTTL call with

> Could not find any Codecs for type Seq[foo.Bar]. Please provide one or import scalacache._

Caffeine is in-memory cache - why is codec i.e. serialization/deserialization needed at all?

**Comment by cb372** [comment]
>Caffeine is in-memory cache - why is codec i.e. serialization/deserialization needed at all?

Good point, the serialization codec is not used for Caffeine (or any of the other in-memory cache implementations). That makes it doubly unfortunate that you got hit by this bug. Sorry about that, and thanks for reporting.

The cause of the problem is that `Seq` does not extend from `java.io.Serializable`, and ScalaCache 0.8.0 only provides codecs for `A <: Serializable`.

I think the best solution is to change ScalaCache's provided codec to support `A <: AnyRef`, and throw an exception at runtime if somebody tries to serialize something that's not `Serializable`. I'll try and get a fix out asap. 

In the meantime, you might be able to work around this by using a concrete type such as `List` instead of `Seq`, because `List` is `Serializable`.

/cc @lloydmeta

**Comment by cb372** [comment]
I've just released 0.8.1, which should fix this issue. Please re-open if that is not the case. Thanks again for reporting!

**Comment by sslavic** [comment]
Both workaround and fix worked well. Thanks you very much!
